operator/v1/types_baremetal.go

```diff
@@ -0,0 +1,68 @@
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+
+// Metal3ProvisioningController contains configuration used by the Provisioning service (Ironic)
+// to provision baremetal hosts.
+//
+type Metal3ProvisioningController struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// spec is the specification of the desired behavior of the
+	// Metal3ProvisioningController.
+	Spec Metal3ProvisioningControllerSpec `json:"spec,omitempty"`
+	// status is the most recently observed status of the
+	// Metal3ProvisioningController.
+	Status Metal3ProvisioningControllerStatus `json:"status,omitempty"`
+}
+
+// ProvisioningDHCP
+type ProvisioningDHCP struct {
+	// managementState indicates whether the DHCP service would be started within
+	// the metal3 cluster or would use an external DHCP service
+	ManagementState ManagementState `json:"managementState"`
+
+	// DHCPRange
+	DHCPRange string `json:"DHCPRange,omitempty"`
+}
+
+// IngressControllerSpec is the specification of the desired behavior of the
+// IngressController.
+type Metal3ProvisioningControllerSpec struct {
+	// provisioningInterface is the name of the network interface on a Baremetal server connected
+	// to the provisioning network.
+	ProvisioningInterface string `json:"provisioningInterface"`
+
+	// provisioningIP is the IP address assigned to the provisioningInterface for provisioning
+	// the baremetal node. This IP address should be within the provisioning subnet, and
```

Comment:
where is this subnet defined, is there validation to ensure this?

Reply:
It will be defined in the installer, and there is WIP (which depends on this PR) to ensure validation of the IP specified:

https://github.com/openshift/installer/pull/2449/

Also related is https://github.com/openshift/installer/issues/2091 and https://github.com/openshift/installer/pull/2320